codegen/src/main/java/software/amazon/awssdk/codegen/poet/batchmanager/BatchFunctionsClassSpec.java

```diff
@@ -109,11 +109,12 @@ private MethodSpec constructor() {
                              .collect(Collectors.toList());
     }
 
-    private Stream<MethodSpec> safeBatchFunctions(Map.Entry<String, BatchManagerMethods> batchFunctions) throws RuntimeException {
+    private Stream<MethodSpec> safeBatchFunctions(Map.Entry<String, BatchManagerMethods> batchFunctions)
+            throws IllegalArgumentException {
         try {
             return batchFunctions(batchFunctions);
         } catch (IllegalArgumentException e) {
-            throw new RuntimeException(e);
+            throw new IllegalArgumentException(e);
         }
```

Comment:
Why do we need to catch IllegalArgumentException to just throw it?

Reply:
Not handling exception here.